libs/isograph-react/src/core/startUpdate.ts

```diff
@@ -1,13 +1,48 @@
-import type {
-  ExtractData,
-  ExtractStartUpdate,
-  UnknownTReadFromStore,
+import {
+  stableIdForFragmentReference,
+  type ExtractStartUpdate,
+  type FragmentReference,
+  type UnknownTReadFromStore,
 } from './FragmentReference';
 import type { IsographEnvironment } from './IsographEnvironment';
 
-export function startUpdate<TReadFromStore extends UnknownTReadFromStore>(
+export function getOrCreateCachedStartUpdate<
+  TReadFromStore extends UnknownTReadFromStore,
+>(
+  environment: IsographEnvironment,
+  fragmentReference: FragmentReference<TReadFromStore, any>,
+  eagerResolverName: string,
+): ExtractStartUpdate<TReadFromStore> {
+  const cachedStartUpdateByResolver = environment.fieldCache;
+
+  const cacheEntry = (cachedStartUpdateByResolver[
+    stableIdForFragmentReference(fragmentReference, eagerResolverName)
+  ] ??= {
+    kind: 'EagerReader',
+    startUpdate: undefined,
+  });
+
+  switch (cacheEntry.kind) {
+    case 'EagerReader': {
+      return (cacheEntry.startUpdate ??= createStartUpdate(
+        environment,
+        fragmentReference,
+      ));
+    }
+    case 'Component': {
+      throw new Error(
```

Comment:
hm. Since we're just throwing if we get the wrong one, maybe we should just have two separate caches. Sorry for going back and forth on this one.

Thoughts?

See #372